grafana-plugin/src/network/oncall-api/http-client.ts

```diff
@@ -4,75 +4,108 @@ import createClient from 'openapi-fetch';
+         * to retry POST requests (and other non-idempotent requests). This doesn't necessarily make these requests
+         * idempotent, but it will make them retry-able from Go's (read: net/http) perspective.
+         *
+         * https://stackoverflow.com/questions/42847294/how-to-catch-http-server-closed-idle-connection-error/62292758#62292758
+         * https://raintank-corp.slack.com/archives/C01C4K8DETW/p1692280544382739?thread_ts=1692279329.797149&cid=C01C4K8DETW
+         */ 'X-Idempotency-Key': `${Date.now()}-${Math.random()}`,
+      },
+    };
 
-    if (!span) {
-      span = tracer.startSpan('http-request');
-      span.setAttribute('page_url', document.URL.split('//')[1]);
-      span.setAttribute(SemanticAttributes.HTTP_URL, url);
-      span.setAttribute(SemanticAttributes.HTTP_METHOD, requestConfig.method);
-    }
+    if (faro && otel) {
+      const tracer = otel.trace.getTracer('default');
+      let span = otel.trace.getActiveSpan();
 
-    return new Promise((resolve, reject) => {
-      otel.context.with(otel.trace.setSpan(otel.context.active(), span), async () => {
-        faro.api.pushEvent('Sending request', { url });
+      if (!span) {
+        span = tracer.startSpan('http-request');
+        span.setAttribute('page_url', document.URL.split('//')[1]);
+        span.setAttribute(SemanticAttributes.HTTP_URL, url);
+        span.setAttribute(SemanticAttributes.HTTP_METHOD, requestConfig.method);
+      }
 
-        try {
-          const response = await fetch(url, {
-            ...requestConfig,
-            headers: {
-              ...requestConfig.headers,
-              /**
-               * In short, this header will tell the Grafana plugin proxy, a Go service which use Go's HTTP Transport,
-               * to retry POST requests (and other non-idempotent requests). This doesn't necessarily make these requests
-               * idempotent, but it will make them retry-able from Go's (read: net/http) perspective.
-               *
-               * https://stackoverflow.com/questions/42847294/how-to-catch-http-server-closed-idle-connection-error/62292758#62292758
-               * https://raintank-corp.slack.com/archives/C01C4K8DETW/p1692280544382739?thread_ts=1692279329.797149&cid=C01C4K8DETW
-               */ 'X-Idempotency-Key': `${Date.now()}-${Math.random()}`,
-            },
-          });
-          faro.api.pushEvent('Request completed', { url });
-          span.end();
-          resolve(response);
-        } catch (error) {
-          faro.api.pushEvent('Request failed', { url });
-          faro.api.pushError(error);
-          span.setStatus({ code: SpanStatusCode.ERROR });
-          span.end();
-          reject(error);
-        }
+      return new Promise((resolve, reject) => {
+        otel.context.with(otel.trace.setSpan(otel.context.active(), span), async () => {
+          faro.api.pushEvent('Sending request', { url });
```

Comment:
we added Faro there a long time ago but seems don't use it to analyze anything...need to discuss do we need it anymore

Reply:
Either way I believe this part can be refactored and simplified, it's a bit of spaghetti here right now. But I wanted to keep what we have for Axios here